frontend/src/chrome/forms/TextInput.tsx

```diff
@@ -12,7 +12,7 @@ interface TextInputProps {
   label?: string
   labelTooltip?: string
   onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void | undefined
-  onChange?: (e: React.ChangeEvent) => void | undefined
+  onChange?: (value: string) => void | undefined
```

Comment:
drop `| undefined`. `onChange` is already optional. Secondly, don't eliminate the `e` event entirely. We need it whenever event propagation needs to be manipulated. Pass the event as an optional second argument instead.

Reply:
This was causing a bug where the input re-renders, causing the cursor to jump to the end of the input.  This approach fixes the bug, but requires that higher-level components just get the value string, not the full onChange event.

Reply:
hrm, ok there's totally something weird going on here, but I'm not confident _only_ changing the argument is a fix. I suspect this strange combination of `useEffect` & `useState` is part of the problem. Let's accept this change so we can keep moving, but let's file an issue titled: `Correct Controlled Component pattern in TextInput`. & like to here: https://reactjs.org/docs/forms.html

Reply:
Done https://github.com/qri-io/qrimatic/issues/196